### PR TITLE
feat: Add dispatcher support to `useUpdateProfileImage` (T-3564)

### PR DESCRIPTION
**Pull request by desfero**
- Adds support for dispatcher when using `useUpdateProfileImage`.
- Updates docs https://docs.lens.xyz/docs/use-update-profile-image

### CODE REVIEW

**Comment by desfero** [comment]
Link T-3564

**Comment by desfero** [comment]
Link T-3564

File: packages/react/src/transactions/adapters/publication-call-gateways/__tests__/PostCallGateway.spec.ts [hunk 1]
```diff
@@ -71,7 +71,7 @@ const mandatoryFallbackMetadata = (request: CreatePostRequest) => ({
   mainContentFocus: PublicationMainFocus[request.contentFocus],
 });
 
-describe(`Given an instance of ${PostCallGateway.name}`, () => {
+describe(`Given an instance of ${CreatePostCallGateway.name}`, () => {
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/publication-call-gateways/__tests__/PostCallGateway.spec.ts:74]: Should we rename the test files for Post and Comment call gw as well?

File: examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ function UploadAndUpdateNewImage({ profile }: { profile: ProfileFieldsFragment }
     error: updateError,
     isPending,
   } = useUpdateProfileImage({
-    profileId: profile.id,
+    profile,
   });
```
Comment (cesarenaldi) [inline examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx:67]: nit: do we need the wrapping object at all or can we pass `profile` directly?

**Review by reecejohnson — APPROVED** [review]
LGTM 👍

File: examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx (hunk 2 above)
Comment (desfero) [reply examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx:67]: personally prefer the wrapping object for the API that's used externally as it's easier to add new fields later without introducing a breaking change. It appears it's also consistent with the hook that we already have: `useUpdateDispatcherConfig`, `useUnfollow`, and others

File: packages/react/src/transactions/adapters/publication-call-gateways/__tests__/PostCallGateway.spec.ts (hunk 1 above)
Comment (desfero) [reply packages/react/src/transactions/adapters/publication-call-gateways/__tests__/PostCallGateway.spec.ts:74]: oh, good catch. Fixed

File: examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx (hunk 2 above)
Comment (cesarenaldi) [reply examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx:67]: I think some... the ones about publication do accept the object directly.

**Pull request merged**